### ISSUE TITLE
ESLint: stylex/valid-styles reports false positives for function parameter references

### ISSUE THREAD

**Issue by skovhus** [facebook/stylex#1520]
### Describe the issue

`stylex/valid-styles` reports false positives for function parameter references in stylex.create arrow functions

The valid-styles rule flags `props.xxx` references inside stylex.create arrow functions as invalid values, even though StyleX compiles these into CSS variables at build time.

### Expected behavior

The rule correctly validates static entries but should allow values inside arrow function bodies in stylex.create, since these are dynamic parameters that StyleX handles at compile time. The function's type annotation already constrains the value — the ESLint rule shouldn't duplicate that check with a much narrower set of allowed values.

This affects all CSS properties where a function parameter is used as the value: `backgroundColor`, `color`, `transitionDelay`, `animationDelay`, `boxShadow`, etc. It also affects nested pseudo-class objects like `":hover": props.glowColor`.

### Steps to reproduce

```tsx
const styles = stylex.create({
  badge: (props: { badgeColor: string }) => ({
    backgroundColor: props.badgeColor, // ❌ "backgroundColor value must be one of: ..."
  }),
});
```
